### PR TITLE
ODSP: Renames and get rid of IDs for trees

### PR DESCRIPTION
**Pull request by vladsud**
Renaming a bunch of structures in contracts.ts to better reflect what they are.
Also get rid of ID on trees as it's not used anywhere, and storage is planning to stop propagating it

### CODE REVIEW

**Comment by jatgarg** [comment]
>                             treeId = odspSnapshot.trees[0].id;

Is spo returning this id?

---
Refers to: packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:875 in b960aaa. [](commit_id = b960aaafda1f8dd1237472b10ca7d4eeb59340e1, deletion_comment = False)

File: packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts [hunk 1]
```diff
@@ -829,13 +828,13 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
         throw new Error("Not implemented yet");
     }
 
-    private initTreesCache(trees: ITree[]) {
+    private initTreesCache(trees: IOdspSnapshotTree[]) {
         trees.forEach((tree) => {
             this.treesCache.set(tree.id, tree);
```
Comment (jatgarg) [inline packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:833]: Will we get this id?

**Comment by vladsud** [comment]
@marcmasmsft please take a look, Jatin is asking good questions - we need to be very clear on semantics meaning.

We receive this payload:

export interface IOdspSnapshot {
    id: string;
    trees: IOdspSnapshotTree[];
    blobs?: IOdspSnapshotBlob[];
    ops?: ISequencedDeltaOpMessage[];
}

Where we rely on presence of IDs here:

export interface IOdspSnapshotTree {
    entries: IOdspSnapshotTreeEntry[];
    id: string;
    sequenceNumber: number;
}

Here, entries could be commit, tree or blob entries.

For tree entries we do not care about IDs:

export interface IOdspSnapshotTreeEntryTree {
    path: string;
    type: "tree";
    // Indicates that this tree entry is unreferenced. If this is not present, the tree entry is considered referenced.
    unreferenced?: true;
}

Other way to show it - we care about red parts, but do not care about green parts:
image

**Comment by vladsud** [comment]
Overall, it's just too complicated for what we are trying to do :)
Can we flatten it and not have commits at all?
Will moving to incremental summaries result in more commits popping up (essentially randomly) in the tree?

**Comment by marcmasmsft** [comment]
Correct, blobs and commits will continue to have ids and those are necessary; +1 on flattening, I'd like to get to that (commit then becomes a tree node) but I'd stage it by first stopping to return ids for trees

---
In reply to: [832265783](https://github.com/microsoft/FluidFramework/pull/6021#issuecomment-832265783) [](http://example.com/codeflow?ancestors=832265783,832265783)

**Comment by marcmasmsft** [comment]
answering the other question about incremental summaries... you wouldn't get extra commit links (even today) unless you are linking the incremental summaries together with commit nodes; I'd suggest it's better to reuse the tree nodes from the commit rather than linking the commit directly (in my future plan, I'll just flatten the commit node and it will become a tree)

---
In reply to: [832278043](https://github.com/microsoft/FluidFramework/pull/6021#issuecomment-832278043) [](http://example.com/codeflow?ancestors=832278043,832265783,832265783)

**Comment by vladsud** [comment]
Super! I've pushed one more commit to rename "trees" to "commits" across couple places, as what we deal with top level are commits, not trees. Using "trees" naming is causing all these confusions. @jtagarg - please take a look

File: packages/drivers/odsp-driver/src/contracts.ts [hunk 2]
```diff
@@ -80,75 +80,93 @@ export interface IDocumentStorageVersion {
...
+    entries: OdspSummaryTreeEntry[];
 }
 
-export interface ISnapshotResponse {
+export interface IWriteSummaryResponse {
     id: string;
 }
 
-export type SnapshotTreeEntry = ISnapshotTreeValueEntry | ISnapshotTreeHandleEntry;
+export type OdspSummaryTreeEntry = IOdspSummaryTreeValueEntry | ISummaryTreeHandleEntry;
```
Comment (jatgarg) [inline packages/drivers/odsp-driver/src/contracts.ts:100]: ISummaryTreeHandleEntry -> IOdspSummaryTreeHandleEntry

File: packages/drivers/odsp-driver/src/contracts.ts [hunk 3]
```diff
@@ -80,75 +80,93 @@ export interface IDocumentStorageVersion {
...
-export interface ISnapshotTreeValueEntry extends ISnapshotTreeBaseEntry {
-    value: SnapshotTreeValue;
+export interface IOdspSummaryTreeValueEntry extends IOdspSummaryTreeBaseEntry {
+    value: OdspSummaryTreeValue;
     // Indicates that this tree entry is unreferenced. If this is not present, the tree entry is considered referenced.
     unreferenced?: true;
 }
 
-export interface ISnapshotTreeHandleEntry extends ISnapshotTreeBaseEntry {
+export interface ISummaryTreeHandleEntry extends IOdspSummaryTreeBaseEntry {
```
Comment (jatgarg) [inline packages/drivers/odsp-driver/src/contracts.ts:113]: ISummaryTreeHandleEntry -> IOdspSummaryTreeHandleEntry

**Review by jatgarg — APPROVED** [review]
:shipit:

**Pull request merged**